### PR TITLE
Th2 3095

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

**Comment by Topru333** [comment]
Are we sure we need to merge into dev instead of master ? We already merged 4c7602d commit into master as i can remember

File: build.gradle [hunk 1]
```diff
@@ -17,15 +17,14 @@
 plugins {
     id 'java'
     id 'maven-publish'
-    id 'org.jetbrains.kotlin.jvm' version '1.3.71'
+    id 'org.jetbrains.kotlin.jvm' version '1.5.21'
```
Comment (ivandruzhinin) [inline build.gradle:20]:
```suggestion
    id 'org.jetbrains.kotlin.jvm' version '1.5.31'
```

File: build.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *  Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
```
Comment (ivandruzhinin) [inline build.gradle:2]:
```suggestion
 *  Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
```

File: src/main/kotlin/com/exactpro/th2/codec/AbstractCodecProcessor.kt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *  Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/AbstractCodecProcessor.kt:2]:
```suggestion
 *  Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
```

File: src/main/kotlin/com/exactpro/th2/codec/DecodeProcessor.kt [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *  Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/DecodeProcessor.kt:2]:
```suggestion
 *  Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
```

File: src/main/kotlin/com/exactpro/th2/codec/EncodeProcessor.kt [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *  Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/EncodeProcessor.kt:2]:
```suggestion
 *  Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
```

File: src/main/kotlin/com/exactpro/th2/codec/api/IPipelineCodecContext.kt [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/api/IPipelineCodecContext.kt:2]:
```suggestion
 * Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
```

File: src/main/kotlin/com/exactpro/th2/codec/api/IPipelineCodecFactory.kt [hunk 7]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/api/IPipelineCodecFactory.kt:2]:
```suggestion
 * Copyright 2021-2022 Exactpro (Exactpro Systems Limited)
```

**Review by ivandruzhinin — APPROVED** [review]

**Review by Topru333 — APPROVED** [review]

File: build.gradle [hunk 8]
```diff
@@ -70,8 +70,8 @@ dependencies {
     implementation 'com.fasterxml.jackson.module:jackson-module-kotlin'
     implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml'
 
-    implementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.5.21'
-    implementation 'org.jetbrains.kotlin:kotlin-reflect:1.5.21'
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:${kotlin_version}"
+    implementation "org.jetbrains.kotlin:kotlin-reflect:${kotlin_version}"
```
Comment (ivandruzhinin) [inline build.gradle:74]: AFAIK you can omit specifying Kotlin dependencies versions - plugin should take care of it

File: src/main/kotlin/com/exactpro/th2/codec/DecodeProcessor.kt [hunk 9]
```diff
@@ -44,7 +47,7 @@ class DecodeProcessor(
             val parentEventId = messageGroup.allParentEventIds
 
             if (messageGroup.messagesList.none(AnyMessage::hasRawMessage)) {
-                parentEventId.onEvent("Message group has no parsed messages in it", messageGroup.messageIds)
+                logger.debug { "Message group has no parsed messages in it" }
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/DecodeProcessor.kt:50]: Why?

File: src/main/kotlin/com/exactpro/th2/codec/EncodeProcessor.kt [hunk 10]
```diff
@@ -42,7 +45,7 @@ class EncodeProcessor(
             val parentEventId = messageGroup.allParentEventIds
 
             if (messageGroup.messagesList.none(AnyMessage::hasMessage)) {
-                parentEventId.onEvent("Message group has no parsed messages in it", messageGroup.messageIds)
+                logger.debug { "Message group has no parsed messages in it" }
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/EncodeProcessor.kt:48]: Why?

**Review by ivandruzhinin — APPROVED** [review]

**Pull request merged**